xdsl/ir.py

```diff
@@ -145,12 +145,16 @@ def name(self) -> str | None:
 
     @name.setter
     def name(self, name: str | None):
-        if name is None or self._name_regex.fullmatch(name):
+        # discard numeric-only names (such as %5)
+        if name is None or name.isnumeric():
```

Comment:
Should we maybe trigger an error here as well? Because this is silently failing (and thus users will be confused if they try to give a numeric name).
Should we rather have only `Printer` making this test?

Reply:
I'm unsure about that, because setting a numeric-only name hint is "allowed" imo, but immediately discarded, right? That's why I didn't throw an error.

Reply:
Maybe we can generate a warning?

Reply:
Maybe we can generate a warning?

Reply:
I mean, the question is mostly: "If you shouldn't call this function with these arguments, should you accept them in the first place?"
I understand that this makes the code easier in the Printer side, but for all rewrites/codegen that are going to call this function, you would like to know that your hint is not discarded, right?

Reply:
Well, my problem with a warning is that it should be treated as an error by the user.

I think we disagree because of a naming problem. From my understanding, you say that an `SSAValue` `name` should allow `5`. However, to me, this field should be renamed `name_hint`, which would then make sense to disallow `5`, since `5` is not an allowed named hint. Would you agree on that?

Reply:
I think we can do some major refactoring around the `SSAValue.name` field. However, I don't want to do them all in this PR.

I think this is more than enough for now, and we can tackle this problem at a later stage.

Reply:
And yes, renaming the field to `name_hint` is on that list of refactorings :D

Reply:
Okay, if you plan on doing this I'm fine, but I would remove the warning nontheless.
This is a precedent I would rather not set, because I really think we should treat warnings as errors.
The only exception I see are deprecated warnings, because this only appears once a user upgrade their code.

Reply:
open an issue on the refactor?

Reply:
done! #585